src/Project.hs

```diff
@@ -382,11 +393,14 @@ wasIntegrationAttemptFor commit pr = case integrationStatus pr of
 
 integratedPullRequests :: ProjectState -> [PullRequestId]
 integratedPullRequests = filterPullRequestsBy $ isIntegrated . integrationStatus
-  where
-  isIntegrated (Integrated _ BuildPending)     = True
-  isIntegrated (Integrated _ (BuildStarted _)) = True
-  isIntegrated (Integrated _ BuildSucceeded)   = True
-  isIntegrated _                               = False
+
+unfailingIntegratedPullRequests :: ProjectState -> [PullRequestId]
+unfailingIntegratedPullRequests = filterPullRequestsBy $ isUnfailingIntegrated . integrationStatus
+
+unfailingIntegratedPullRequestsBefore :: PullRequest -> ProjectState -> [PullRequestId]
+unfailingIntegratedPullRequestsBefore referencePullRequest = filterPullRequestsBy $
+  \pr -> isUnfailingIntegrated (integrationStatus pr)
+      && referencePullRequest `approvedAfter` pr
```

Comment:
`nonFailedPullRequests`?

Reply:
> `nonFailedPullRequests`?

I like it, but it misses the "integrated" part. I don't know how important that is to mention in the name though.

Reply:
Umm... you are right.  I will use [unfailed] instead of [unfailing]().

[unfailing]: https://en.wiktionary.org/wiki/unfailing
[unfailed]: https://en.wiktionary.org/wiki/unfailed